api/modules/publishedProjects/controller.js

```diff
@@ -0,0 +1,55 @@
+var Boom = require('boom');
+
+var Errors = require('../../classes/errors');
+var BaseController = require('../../classes/base_controller');
+var Model = require('./model');
+var Project = require('../projects/model');
+var File = require('../files/model');
+var controller = new BaseController(Model);
+
+function copyFile(file, projectId) {
+  return File.forge({
+    path: file.get('path'),
+    project_id: projectId,
+    buffer: file.get('buffer')
+  }).save()
+  .then(function(projectFile) {
+    if(!projectFile) { throw Boom.notFound(); }
```

Comment:
Is this block necessary?  AFAIK If the save fails an error will be thrown automatically. 

Likewise, if you don't want the projectfile returned to the parent promise, just:

```js
return Promise.resolve().then(function() {
  File.forge({}).save();
});
```